### PR TITLE
Linking User (setter) to climb form + adding userID in db for Climbs collection

### PR DESCRIPTION
**Pull request by VGmt**
- Adding User details to climb creation form (user.email)
- Removing call for all existing climbs in database in ClimbsInputData (why was that there in the first place is a mystery)
- Renamed Authentication to SignInUpToggle, to make it clearer what it does
- Adding React Context file above App parent compo to make it accessible everywhere

WhatsApp Image 2023-10-05 at 5 13 07 PM

### CODE REVIEW

File: trk/src/App.js [hunk 1]
```diff
@@ -1,30 +1,29 @@
...
 
-function App(props) {
-  const [initializing, setInitializing] = useState(true);
-  const [user, setUser] = useState();
 
-  function onAuthStateChanged(user) {
-    setUser(user);
-    if (initializing) setInitializing(false);
-  }
+function AppWrapper() {
```
Comment (VGmt) [inline trk/src/App.js:9]: To make the User 'context' available throughout the code (i.e. access user data evrywhr), I had to include an Wrapper around the app. For some reason, wrapping the <App /> component in <AuthProvider /> alone wasn't enough (down below). Idk why, but I needed to create a wrapper around the App function.

File: trk/src/App.js [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
-  }, []);
+function App(props) {
 
-  if (initializing) return null;
+  const { currentUser } = useContext(AuthContext);
 
   return (
     <PaperProvider>
-      {user ? <AppNavigator /> : <AuthNavigator />}
+      {currentUser ? <AppNavigator /> : <AuthNavigator />}
```
Comment (VGmt) [inline trk/src/App.js:24]: Here

File: trk/src/Components/SignInUpToggle.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { View, Button, StyleSheet } from 'react-native';
 import SignIn from './SignIn';
 import SignUp from './SignUp';
 
-const Authentication = () => {
+const SignInUpToggle = () => {
```
Comment (VGmt) [inline trk/src/Components/SignInUpToggle.js:6]: Renamed Auth -> SignInUpToggle

File: trk/src/api/ClimbsApi.js [hunk 4]
```diff
@@ -3,33 +3,9 @@ import React from "react";
 import { useState, useEffect } from "react";
 
 function ClimbsApi() {
-    const [climbs, setClimbs] = useState([]);
-    const [loading, setLoading] = useState(false);
 
     const ref = firebase.firestore().collection("climbs");
 
-    // Realtime subscription
```
Comment (VGmt) [inline trk/src/api/ClimbsApi.js:11]: Removing crazy db calls

File: trk/src/api/ClimbsApi.js [hunk 5]
```diff
@@ -39,10 +15,8 @@ function ClimbsApi() {
     }
 
     return {
-        climbs,
-        loading,
         addClimb,
-        getClimb,
+        getClimb
```
Comment (VGmt) [inline trk/src/api/ClimbsApi.js:19]: removing return info that we never used

File: trk/src/Screens/ClimbList/index.js [hunk 6]
```diff
@@ -90,6 +99,12 @@ const ClimbInputData = () => {
           onChangeText={setImage}
           placeholder="Enter image"
         />
+
```
Comment (VGmt) [inline trk/src/Screens/ClimbList/index.js:102]: user info in form

File: trk/src/Screens/ClimbList/index.js [hunk 7]
```diff
@@ -7,21 +7,30 @@ import writeSignature from "../../NfcUtils/writeSignature";
...
 
-  const { addClimb, getClimb } = ClimbsApi();
 
   function handleAddClimb() {
     const climb = {
       name,
       grade,
       location,
       image,
+      setter: setter.uid
```
Comment (VGmt) [inline trk/src/Screens/ClimbList/index.js:33]: 'setter' = a whole of info on the user. You have to select the .uid extension to get the user ID

**Review by redpepper-nag — APPROVED** [review]
seems good

**Pull request merged**